pkg/server/endpoints/node/handler.go

```diff
@@ -480,7 +488,70 @@ func (h *Handler) FetchJWTSVID(ctx context.Context, req *node.FetchJWTSVIDReques
 }
 
 func (h *Handler) PushJWTKeyUpstream(ctx context.Context, req *node.PushJWTKeyUpstreamRequest) (resp *node.PushJWTKeyUpstreamResponse, err error) {
-	return nil, status.Error(codes.Unimplemented, "cannot push JWK upstream")
+	counter := telemetry_server.StartNodeAPIPushJWTKeyUpstreamCall(h.c.Metrics)
+	defer counter.Done(&err)
+	log := h.c.Log.WithField(telemetry.Method, telemetry.PushJWTKeyUpstream)
+
+	_, ok := getPeerCertificate(ctx)
+	if !ok {
+		log.Error("Downstream SVID is required for this request")
+		return nil, status.Error(codes.InvalidArgument, "downstream SVID is required for this request")
+	}
+
+	_, ok = getDownstreamEntry(ctx)
+	if !ok {
+		log.Error("Downstream entry is required for this request")
+		return nil, status.Error(codes.InvalidArgument, "downstream entry is required for this request")
+	}
+
+	err = h.limiter.Limit(ctx, PushJWTKey, 1)
+	if err != nil {
+		log.WithError(err).Error("Rejecting request due to JWK push rate limiting")
+		return nil, status.Error(codes.ResourceExhausted, err.Error())
+	}
+
+	upstreamAuth, useUpstream := h.c.Catalog.GetUpstreamAuthority()
+	if useUpstream { // We are an intermediate SPIRE server
+		publishCtx, cancel := context.WithTimeout(ctx, h.c.PublishJWKTimeout)
+		defer cancel()
+		res, err := upstreamAuth.PublishJWTKey(
+			publishCtx,
+			&upstreamauthority.PublishJWTKeyRequest{
+				JwtKey: req.JwtKey,
+			})
+
+		switch {
+		case gcu.IsUnimplementedError(err):
+			h.warnOnce.Do(func() {
+				log.Warn("UpstreamAuthority does not support JWT-SVIDs. Workloads managed " +
+					"by this server may have trouble communicating with workloads outside " +
+					"this cluster when using JWT-SVIDs.")
+			})
+		case err != nil:
+			return nil, status.Error(codes.Internal, err.Error())
+		default:
+			// Send back to the caller the received JWKs from upstream,
+			// this will change soon according to https://github.com/spiffe/spire/issues/1372#issuecomment-589469449
+			return &node.PushJWTKeyUpstreamResponse{
+				JwtSigningKeys: res.UpstreamJwtKeys,
+			}, nil
+		}
+	}
+
+	if err := h.c.Manager.AppendBundle(ctx, nil, []*common.PublicKey{req.JwtKey}); err != nil {
```

Comment:
I wonder if this call method can return the appended bundle. Then we can just return it directly and not bother with the extra call to the datastore.

We might still want to evict the cache since we know it changed.

_Wonders if we should have provided a way for the manager to invalidate the node cache.... anyway that can wait for another PR_

Reply:
> I wonder if this call method can return the appended bundle. Then we can just return it directly and not bother with the extra call to the datastore.

Yes it can! Thanks :smiley: 

> We might still want to evict the cache since we know it changed.

That's right.

> Wonders if we should have provided a way for the manager to invalidate the node cache.... anyway that can wait for another PR

Maybe...

> anyway that can wait for another PR

:tada: